ui/src/app/shared/edge/edgeconfig.ts

```diff
@@ -157,6 +157,40 @@ export class EdgeConfig {
         }
     }
 
+    /**
+     * Determines if Edge has a producing device
+     * 
+     */
+    public hasProducer(): boolean {
+        if (this.getComponentsImplementingNature("io.openems.edge.meter.api.SymmetricMeter").filter(component => {
+            if (component.isEnabled == false) {
+                return false;
+            }
+            else if (component.properties['type'] == "PRODUCTION") {
+                return true;
+            } else {
+                // TODO make sure 'type' is provided for all Meters
+                switch (component.factoryId) {
+                    case 'Fenecon.Mini.PvMeter':
+                    case 'Fenecon.Dess.PvMeter':
+                    case 'Fenecon.Pro.PvMeter':
+                    case 'Kostal.Piko.Charger':
+                    case 'Kaco.BlueplanetHybrid10.PvInverter':
+                    case 'PV-Inverter.Solarlog':
+                    case 'PV-Inverter.KACO.blueplanet':
+                    case 'PV-Inverter.SunSpec':
+                    case 'SolarEdge.PV-Inverter':
+                        return true;
+                }
+            }
+        }).length > 0 || this.getComponentsImplementingNature('io.openems.edge.ess.dccharger.api.EssDcCharger').length > 0) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
```

Comment:
/**
     * Determines if Edge has a producing device
     */
    public hasProducer(): boolean {
        // Do we have a Ess DC Charger?
        if(this.getComponentsImplementingNature('io.openems.edge.ess.dccharger.api.EssDcCharger').length > 0) {
            return true;
        }
        // Do we have a Meter with type PRODUCTION?
        for(let component of this.getComponentsImplementingNature("io.openems.edge.meter.api.SymmetricMeter")) {
            if (component.isEnabled) {
                // TODO make sure 'type' is provided for all Meters
                if (component.properties['type'] == "PRODUCTION") {
                    return true;
                }
                switch (component.factoryId) {
                    case 'Fenecon.Mini.PvMeter':
                    case 'Fenecon.Dess.PvMeter':
                    case 'Fenecon.Pro.PvMeter':
                    case 'Kostal.Piko.Charger':
                    case 'Kaco.BlueplanetHybrid10.PvInverter':
                    case 'PV-Inverter.Solarlog':
                    case 'PV-Inverter.KACO.blueplanet':
                    case 'PV-Inverter.SunSpec':
                    case 'SolarEdge.PV-Inverter':
                        return true;
                }
            }
        }
        return false;
    }